rfcs/0000-dns-cluster-aware-delegation.md

```diff
@@ -0,0 +1,170 @@
+# DNS Cluster Aware Delegation
+
+- Feature Name: `dns_cluster_aware_delegation`
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/pull/0000)
+- Issue tracking: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Proposal to add functionality to delegate the processing of a DNSRecord to a designated cluster or clusters in a multi cluster environment. 
+
+# Motivation
+[motivation]: #motivation
+
+Multi cluster DNS is currently achieved by using the eventual provider DNS service (AWS Route etc ..) as a store for ownership metadata using specially created TXT records, and as a central API service that all clusters can communicate with.
+Each cluster processes its own DNSRecords, becoming aware of other DNSRecords contributing to the same set of endpoints via this centrally stored data, in turn allowing it to correctly translate the DNSRecord endpoints into an appropriate API operation (Create/Update/Delete) and get to the desired state. 
+
+In some cases, such as our current CoreDNS solution, there is no central off cluster API service that can be configured on all clusters and as such a different approach to multiple cluster discovery and reconciliation is required.
+
+* Add a provider agnostic alternative for multi cluster dns support while still maintaining current behaviour for existing providers.
+* Reduce the number of required CoreDNS instances to one, but still allow multiple if required, redundancy etc...
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## DNS Operator modes
+
+The DNS Operator can be configured to work in different modes specified via a controller flag `mode` which must be one of `default`, `primary` or `remote` i.e. `--mode=[default|primary|remote]`.
+To allow for customisation of this mode via an OLM deployment of the kuadrant-operator, the mode will also be configurable via an environment variable `DNS_MODE`.  
```

Comment:
I think it is fine to have the env var, but we should still consider surfacing the dns mod via the kuadrant CR

Reply:
I was thinking if with kuadrant we provided a configmap that the DNSOpertor used for env vars. If installing the dns operator yourself you can just set the env var, but with kuadrant we provide a configmap that the kuadrant operator could then use to configure the DNS Operator?

Reply:
You can do it via a subscription, but I think a configmap is a more generic approach as you say. No it doesn't need to be done as part of this work but is something we should likely capture